Algorithms/Search/exponential_search.cs

```diff
@@ -0,0 +1,74 @@
+// C# program to find an element x in a 
+// sorted array using Exponential search. 
+using System; 
+class GFG { 
+  
+// Returns position of first 
+// occurrence of x in array 
+static int exponentialSearch(int []arr,  
+                         int n, int x) 
+{ 
+      
+    // If x is present at  
+    // first location itself 
+    if (arr[0] == x) 
+        return 0; 
+  
+    // Find range for binary search  
+    // by repeated doubling 
+    int i = 1; 
+    while (i < n && arr[i] <= x) 
+        i = i * 2; 
+  
+    // Call binary search for 
+    // the found range. 
+    return binarySearch(arr, i/2,  
+                       Math.Min(i, n), x); 
+} 
+  
+// A recursive binary search 
+// function. It returns location 
+// of x in given array arr[l..r] is 
+// present, otherwise -1 
+static int binarySearch(int []arr, int l, 
```

Comment:
Codacy found an issue: [Rename method 'binarySearch' to match pascal case naming rules, consider using 'BinarySearch'.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6263297)